### DISCUSSION TITLE
Problem with running simulation on Pmetrics version 2.0.2

### DISCUSSION THREAD

**Discussion by SDasliv** [LAPKB/Pmetrics#162] [Q&A]
I'm receiving an error message when trying to run a simulation from a population PK model, that was working on the previous version of pmetrics. The code I am using is:
SIMrun(final.35,model="xxx.txt",data="xxx.csv",nsim=5000,limits=NA,split=T, predInt = 1,outname="simout1")
I receive the following:
Error in `[.data.frame`(x, order(x, na.last = na.last, decreasing = decreasing)) : 
  undefined columns selected
In addition: Warning messages:
1: In stringr::str_split(err[i + 1], ",")[[1]] %>% as.numeric() :
  NAs introduced by coercion
2: In xtfrm.data.frame(x) : cannot xtfrm data frames
3: In xtfrm.data.frame(x) : cannot xtfrm data frames

I can run the simulation if I just use the mean and CV from the model.  I have tried both by loading the model which was generated in the previous version of pmetrics, and also by re-running the model again in V2.0.2. 

Any suggestions please?

**Comment by mhovd** [comment]
I am not sure, but this could be related to #127. 
Do you have any non-numeric IDs in your template (`xxx.csv`)? If so, you could try to replace those with numeric only IDs.

**Comment by Hector-Perez-Madrid** [comment]
I had a very similar problem here

I have loaded a model created with Pmetrics 1.5.1 to do a simulation in Pmetrics 2.0

I was using the following code:
SIMrun(poppar=final.3,limits=NA, model="model.txt", data="data.csv",predInt=0.25)
-**The IDs in the template are numeric**-

  Error reported was:
Error in `[.data.frame`(x, order(x, na.last = na.last, decreasing = decreasing)) :
   undefined columns selected
In addition: Warning messages:
1: In xtfrm.data.frame(x) : cannot xtfrm data frames
2: In xtfrm.data.frame(x) : cannot xtfrm data frames


**But**, when i used PM_sim$run, it worked!

simul<-PM_sim$run(poppar=run3,limits=NA, model="model.txt", data="data.csv",predInt=0.25)

**Reply by mnneely** [reply]
@Hector-Perez-Madrid, thanks for your reply. Since you are running your simulation based on a prior model run, you can take advantage of the `$sim()` method attached to `PM_result` objects. Your method will work, but using PM_sim$run() is really meant for manual `poppar` objects. 

```
run3 <- PM_load(3)
simul <- run3$sim(limits=NA, data="data.csv",predInt=0.25)
```
This assumes that your model is the same as the model in run3. Note that the poppar (run3$final) is automatically supplied for you. If you were using the same data as for run3, you wouldn't even need to supply the data.
